### PR TITLE
Facilities summarys and Fac-arrays

### PR DESCRIPTION
**Pull request by Bane70**
This should be working fine. The summaries do need a bit more work eventually I think, as 'SA get milked' is usually not being displayed currently, as to where it has to be for it to display things, I might play around with things a bit later on to correct the summaries.

### CODE REVIEW

**Comment by Bane70** [comment]
Well CCeaner just wiped out my saves for the game so I'm off to start a new one, Or NG+ if I can find an old save on the HD. NG+ didn't get tested but should work, maybe I'll find out soon enough. So further testing on this is a bit limited by me at the moment.

**Comment by Bane70** [comment]
You need to do a new game or NG+

**Comment by Scriptifaber** [comment]
@Bane70 yep, you're right. Looks fine in a new game.

**Comment by Folant** [comment]
The Arcade/Brothel weekly earnings seem to be broken.

**Comment by Bane70** [comment]
Thanks. Yeah I kinda thought it might do that to a couple of facilities. Are you sure about the Brothel though? as it should actually be giving the amounts in their individual summaries, I could change it around for facilities to display the incomes at the end of the slave summaries (the second paragraph), That would also allow for the 'SA get milked' to be displayed.
So it would be displaying the info about the Facility leader if present, Then a bit about the facility how many workers etc. Then display the slave info's and then any incomes/whatnot.
Or for places like the arcade/brothel etc. I could capture the added amounts from the individuals and display it at the end.

**Comment by Folant** [comment]
>. Are you sure about the Brothel though? 

Well, I guess it shouldn't be like that. The Arcade gives a negative profit sometimes, lol
br
h

**Comment by Bane70** [comment]
Yeah I agree it shouldn't be like that. I could also go into the Sa's and change the texts to be put in a variable to be printed out later. Though knowing twine it would probably mess things up later with the amounts with the way it displays variables.
So I think I'll rearrange things to have the facility incomes calculated as before and display it at the end of the Fac. report. Hmm or if I can figure out span/replace.

**Comment by Folant** [comment]
Profit looks good now. If the quantity of products is now shown, what about these zeros?
iii

**Comment by Bane70** [comment]
Oh I also forgot to mention from the first set of screenshots it looks like there another error in the texts, as it says "Her asshole recieves constant use and loosens". Then 5 sentences later "her gaping anus recovers a little". Which is in 'SA long term effects' I think.

**Comment by Bane70** [comment]
Hmm I'm not sure about that one. 'SA get milked' Was it her first week maybe?

**Comment by Folant** [comment]
>Was it her first week maybe?

No, it's not. Tested and got this several weeks in a row.

**Comment by Bane70** [comment]
Maybe the problem is in the JavaScript?
<< set $milk = **milkAmount($slaves[$i])**>>
Or maybe your $servantMilkersMultiplier got set to zero somehow << set $milk *= $servantMilkersMultiplier>>

**Comment by Folant** [comment]
Lol maybe. I can offer to wrap this in saGetMilked
````As a result, she produces $milk liters of milk over the week.
	<<if ($arcologies[0].FSPastoralistLaw == 1)>>
		<<set $milkSale to $milk*(9+Math.trunc($arcologies[0].FSPastoralist/30))>>
		Since breast milk is $arcologies[0].name's only legal dairy product, she can scarcely be milked fast enough, and she makes @@color:yellowgreen;¤$milkSale.@@
	<<elseif ($arcologies[0].FSPastoralist != "unset")>>
		<<set $milkSale to $milk*(7+Math.trunc($arcologies[0].FSPastoralist/30))>>
		Since milk is fast becoming a major part of the $arcologies[0].name's dietary culture, her milk is in demand, and she makes @@color:yellowgreen;¤$milkSale.@@
	<<else>>
		<<set $milkSale to $milk*8>>
		Her milk is sold for @@color:yellowgreen;¤$milkSale.@@
	<</if>>
	<<set $cash += $milkSale>>
```` 
with
`<<if $milk > 0>>
    ...
<</if>>`

**Comment by Bane70** [comment]
Hmm. maybe if she just produced 1 then comes to this bit of code it'll become zero.
Might want to be using math.Round as the 0.5 would get rounded up instead of truncated.
	<<elseif ($slaves[$i].assignment is "be confined in the arcade")>>
		<<set $milk *= 0.5>>
	<</if>>

	<<set $milk *= $servantMilkersMultiplier>>

	<<set $milk = Math.trunc($milk)>>
	<<set $slaves[$i].milk += $milk>>
	<<set $milkTotal += $milk>>

	As a result, she produces $milk liters of milk over the week.

	<<if ($arcologies[0].FSPastoralistLaw == 1)>>

**Comment by Bane70** [comment]
with
<<if $milk > 0>> ... <</if>>

And do the same for cum.

Yeah or <<if $milk < 1>><<set $milk = 1>>. ditto for cum. just after the trunc($milk)

**Comment by Folant** [comment]
git gud gets new meaning for sure

**Comment by Bane70** [comment]
OK, I think I got everything updated proper now. Except for maybe a missing/extra extra <br> here and there.

**Comment by Bane70** [comment]
Well as I been playing a bit I see that slaves in facilities aren't managing their diets and such. Lol their just starving away etc. whatever they were set to when I placed them into the facility.
Is their a way to correct this, short of writing a widget to manage things, Or having the RA also do basic stuffs there as well? Should I of have also put in 'saRules'. OK going through it that wouldn't fix anything.

**Comment by Bane70** [comment]
Seems to be more jobs that have to be changed over in RESS.tw and other re*.tw files, and saChooseownJob.tw

Going to get some sleep then work on them in the afternoon. Should be easy to do with the widgets now.

File: src/uncategorized/defaultRules.tw [hunk 1]
```diff
@@ -1275,42 +1275,31 @@ consequences.
...
-		<<set $activeSlave.assignment to "live with your Head Girl">>
-		<<set $activeSlave.assignmentVisible to 0>>
+	<<if ($activeSlave.indentureRestrictions <= 0) && ($activeSlave.assignment != "live with your Head Girl")>>
 		<br>$activeSlave.slaveName has automatically been assigned to $activeSlave.assignment.
 		<<if ($personalAttention is $activeSlave.ID)>>
 			$activeSlave.slaveName no longer has your personal attention; you plan to focus on business.
-			<<set $personalAttention to "business">>
 		<</if>>
-	<</if>>
+		<<assginJob $activeSlave _currentRule.assignFacility>>
```
Comment (Folant) [inline src/uncategorized/defaultRules.tw]: Did you mean `assignJob` everywhere?

Comment (Bane70) [inline src/uncategorized/defaultRules.tw]: No I don't think so oops. Lol Thanks.

File: src/uncategorized/slaveAssignmentsReport.tw [hunk 2]
```diff
@@ -83,168 +83,167 @@
...
-	<<set $Concubine to $slaves[_i], $fuckSlaves++>>
+	<<set $Concubine = $slaves[_i], $fuckSlaves++>>
 <<case "serve in the master suite" "please you">>
 	<<set $fuckSlaves++>>
 <</switch>>
 
 <<if $Lurcher>>
-	<<if $slaves[_i].ID is $Lurcher.ID>>
-		<<set $Lurcher to $slaves[_i]>>
+	<<if $slaves[_i].ID = $Lurcher.ID>>
```
Comment (Folant) [inline src/uncategorized/slaveAssignmentsReport.tw]: Shouldn't there be `==`?

Comment (Bane70) [inline src/uncategorized/slaveAssignmentsReport.tw]: Yes it should! I thought I fixed that before I copied the file here. Thanks again.

File: src/uncategorized/defaultRules.tw (hunk 1 above)
Comment (Bane70) [inline src/uncategorized/defaultRules.tw]: Oh well I was looking at the /if not the widget name. lol. Gotta hate insomnia at times. Turns me into a zombie that is about only capable of doing repetitive routines.

**Comment by Bane70** [comment]
This seems stable now, for me at least. code has been cleaned up and except for my misspelled widget all seems fine. Though I would like to be able to get one of those events to make sure that part is working right, without being in cheat mode to get it.

**Pull request merged**